lib/bigquery/table.js

```diff
@@ -346,8 +364,8 @@ Table.prototype.createWriteStream = function(metadata) {
  * @example
  * table.delete(function(err, apiResponse) {});
  */
-Table.prototype.delete = function(callback) {
-  this.makeReq_('DELETE', '', null, null, callback);
+Table.prototype.delete = function() {
+  return ServiceObject.prototype.delete.apply(this, arguments);
```

Comment:
This is the first time I've seen a `return`, I know we don't really need it since everything is async, but could we be consistent with either returning or not returning across the board?